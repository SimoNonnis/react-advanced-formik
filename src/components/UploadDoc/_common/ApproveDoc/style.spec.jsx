/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import noop from 'lodash/noop';

import { theme } from '@housesimple/react-components';

import {
  PreviewContainer,
  Preview,
  FileName,
  DocPlaceholder,
  FilePreview,
  ConfirmContainer,
  ConfirmLabel,
  ButtonsContainer,
  Button,
} from './style';

describe('Test <PreviewContainer />', () => {
  it('should render PreviewContainer with theme props', () => {
    const wrapper = shallow(<PreviewContainer theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <Preview />', () => {
  it('should render Preview with theme props', () => {
    const wrapper = shallow(<Preview theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <FileName />', () => {
  it('should render FileName with theme props', () => {
    const wrapper = shallow(<FileName theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <DocPlaceholder />', () => {
  it('should render DocPlaceholder with theme props', () => {
    const wrapper = shallow(<DocPlaceholder theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <FilePreview />', () => {
  it('should render FilePreview with theme props', () => {
    const wrapper = shallow(<FilePreview theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <ConfirmContainer />', () => {
  it('should render ConfirmContainer with theme props', () => {
    const wrapper = shallow(<ConfirmContainer theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <ConfirmLabel />', () => {
  it('should render ConfirmLabel with theme props', () => {
    const wrapper = shallow(<ConfirmLabel theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <ButtonsContainer />', () => {
  it('should render ButtonsContainer with theme props', () => {
    const wrapper = shallow(<ButtonsContainer theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <Button />', () => {
  it('should render Button with theme props', () => {
    const wrapper = shallow(
      <Button theme={theme} onClick={noop}>
        Test
      </Button>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
