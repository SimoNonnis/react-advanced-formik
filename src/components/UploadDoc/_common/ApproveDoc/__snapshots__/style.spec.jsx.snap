// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button /> should render Button with theme props 1`] = `
<Button
  className="sc-gqjmRU kwvzZc"
  disabled={false}
  fullWidth={false}
  loading={false}
  onClick={[Function]}
  rounded={3}
  size=""
  styleType=""
  theme={
    Object {
      "base": Object {
        "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "fontSize": "14px",
        "fontWeight": 400,
        "lineHeight": 1.45,
      },
      "breakpoints": Array [
        504,
        671,
        867,
        1050,
        1200,
      ],
      "color": Object {
        "black": "#000000",
        "coral": "#F4364C",
        "darkGrey": "#77838d",
        "green": "#35A856",
        "lightBlue1": "#daf3ff",
        "lightBlue2": "#bfe8f8",
        "lightGreen": "#B4E2C2",
        "lightGrey1": "#d6d6d6",
        "lightGrey2": "#e6e6e6",
        "lightGrey3": "#f6f6f6",
        "lightGrey4": "#c8cccf",
        "lightOrange": "#ffeaca",
        "lightPink": "#f8d7e0",
        "mediumBlue": "#0099cf",
        "mediumGrey": "#979797",
        "orange": "#ffad2f",
        "pink": "#e46185",
        "red": "#f80000",
        "red15": "#fedede",
        "stoneGrey": "#a6adb4",
        "teal": "#06A693",
        "veryDarkBlue": "#1D3141",
        "veryDarkBlue20": "#C8CCCF",
        "veryDarkBlue40": "#9DA5AB",
        "veryDarkBlue60": "#717D87",
        "veryDarkBlue7": "#EEF0F3",
        "veryDarkBlue80": "#4A5A67",
        "white": "#ffffff",
      },
      "maxWidth": "1200px",
      "space": Array [
        5,
        10,
        15,
        20,
        30,
      ],
      "spacing": Object {
        "lg": "28px",
        "md": "18px",
        "sm": "10px",
        "xl": "36px",
        "xs": "5px",
      },
      "typography": Object {
        "caption": Object {
          "desktop": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
          "phone": Object {
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.571rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
        },
        "h1": Object {
          "desktop": Object {
            "fontSize": "2.286rem",
            "lineHeight": "2.571rem",
          },
          "phone": Object {
            "fontSize": "1.714rem",
            "fontWeight": 600,
            "lineHeight": "2rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "2rem",
            "lineHeight": "2.429rem",
          },
        },
        "h2": Object {
          "desktop": Object {
            "fontSize": "1.571rem",
            "lineHeight": "1.857rem",
          },
          "phone": Object {
            "fontSize": "1.429rem",
            "fontWeight": 600,
            "lineHeight": "1.714rem",
            "margin": "0 0 18px 0",
          },
          "tablet": Object {
            "fontSize": "1.429rem",
            "lineHeight": "1.714rem",
          },
        },
        "h3": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 700,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
        "intro": Object {
          "desktop": Object {
            "fontSize": "1.357rem",
            "lineHeight": "1.929rem",
          },
          "phone": Object {
            "fontSize": "1.214rem",
            "fontWeight": 300,
            "lineHeight": "1.714rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "1.214rem",
            "lineHeight": "1.714rem",
          },
        },
        "p": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 400,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
      },
    }
  }
  type="submit"
>
  Test
</Button>
`;

exports[`Test <ButtonsContainer /> should render ButtonsContainer with theme props 1`] = `
<div
  className="sc-gZMcBi geojgz"
/>
`;

exports[`Test <ConfirmContainer /> should render ConfirmContainer with theme props 1`] = `
<div
  className="sc-htoDjs dwVlIy"
/>
`;

exports[`Test <ConfirmLabel /> should render ConfirmLabel with theme props 1`] = `
<span
  className="sc-iwsKbI uTWQM"
/>
`;

exports[`Test <DocPlaceholder /> should render DocPlaceholder with theme props 1`] = `
<DocumentPlaceholder
  className="sc-EHOje kyfnqM"
  fill="currentColor"
  height={40}
  theme={
    Object {
      "base": Object {
        "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "fontSize": "14px",
        "fontWeight": 400,
        "lineHeight": 1.45,
      },
      "breakpoints": Array [
        504,
        671,
        867,
        1050,
        1200,
      ],
      "color": Object {
        "black": "#000000",
        "coral": "#F4364C",
        "darkGrey": "#77838d",
        "green": "#35A856",
        "lightBlue1": "#daf3ff",
        "lightBlue2": "#bfe8f8",
        "lightGreen": "#B4E2C2",
        "lightGrey1": "#d6d6d6",
        "lightGrey2": "#e6e6e6",
        "lightGrey3": "#f6f6f6",
        "lightGrey4": "#c8cccf",
        "lightOrange": "#ffeaca",
        "lightPink": "#f8d7e0",
        "mediumBlue": "#0099cf",
        "mediumGrey": "#979797",
        "orange": "#ffad2f",
        "pink": "#e46185",
        "red": "#f80000",
        "red15": "#fedede",
        "stoneGrey": "#a6adb4",
        "teal": "#06A693",
        "veryDarkBlue": "#1D3141",
        "veryDarkBlue20": "#C8CCCF",
        "veryDarkBlue40": "#9DA5AB",
        "veryDarkBlue60": "#717D87",
        "veryDarkBlue7": "#EEF0F3",
        "veryDarkBlue80": "#4A5A67",
        "white": "#ffffff",
      },
      "maxWidth": "1200px",
      "space": Array [
        5,
        10,
        15,
        20,
        30,
      ],
      "spacing": Object {
        "lg": "28px",
        "md": "18px",
        "sm": "10px",
        "xl": "36px",
        "xs": "5px",
      },
      "typography": Object {
        "caption": Object {
          "desktop": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
          "phone": Object {
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.571rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
        },
        "h1": Object {
          "desktop": Object {
            "fontSize": "2.286rem",
            "lineHeight": "2.571rem",
          },
          "phone": Object {
            "fontSize": "1.714rem",
            "fontWeight": 600,
            "lineHeight": "2rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "2rem",
            "lineHeight": "2.429rem",
          },
        },
        "h2": Object {
          "desktop": Object {
            "fontSize": "1.571rem",
            "lineHeight": "1.857rem",
          },
          "phone": Object {
            "fontSize": "1.429rem",
            "fontWeight": 600,
            "lineHeight": "1.714rem",
            "margin": "0 0 18px 0",
          },
          "tablet": Object {
            "fontSize": "1.429rem",
            "lineHeight": "1.714rem",
          },
        },
        "h3": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 700,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
        "intro": Object {
          "desktop": Object {
            "fontSize": "1.357rem",
            "lineHeight": "1.929rem",
          },
          "phone": Object {
            "fontSize": "1.214rem",
            "fontWeight": 300,
            "lineHeight": "1.714rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "1.214rem",
            "lineHeight": "1.714rem",
          },
        },
        "p": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 400,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
      },
    }
  }
  width={40}
/>
`;

exports[`Test <FileName /> should render FileName with theme props 1`] = `
<P
  className="sc-ifAKCX wFiJQ"
  theme={
    Object {
      "base": Object {
        "fontFamily": "\\"proxima-nova\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "fontSize": "14px",
        "fontWeight": 400,
        "lineHeight": 1.45,
      },
      "breakpoints": Array [
        504,
        671,
        867,
        1050,
        1200,
      ],
      "color": Object {
        "black": "#000000",
        "coral": "#F4364C",
        "darkGrey": "#77838d",
        "green": "#35A856",
        "lightBlue1": "#daf3ff",
        "lightBlue2": "#bfe8f8",
        "lightGreen": "#B4E2C2",
        "lightGrey1": "#d6d6d6",
        "lightGrey2": "#e6e6e6",
        "lightGrey3": "#f6f6f6",
        "lightGrey4": "#c8cccf",
        "lightOrange": "#ffeaca",
        "lightPink": "#f8d7e0",
        "mediumBlue": "#0099cf",
        "mediumGrey": "#979797",
        "orange": "#ffad2f",
        "pink": "#e46185",
        "red": "#f80000",
        "red15": "#fedede",
        "stoneGrey": "#a6adb4",
        "teal": "#06A693",
        "veryDarkBlue": "#1D3141",
        "veryDarkBlue20": "#C8CCCF",
        "veryDarkBlue40": "#9DA5AB",
        "veryDarkBlue60": "#717D87",
        "veryDarkBlue7": "#EEF0F3",
        "veryDarkBlue80": "#4A5A67",
        "white": "#ffffff",
      },
      "maxWidth": "1200px",
      "space": Array [
        5,
        10,
        15,
        20,
        30,
      ],
      "spacing": Object {
        "lg": "28px",
        "md": "18px",
        "sm": "10px",
        "xl": "36px",
        "xs": "5px",
      },
      "typography": Object {
        "caption": Object {
          "desktop": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
          "phone": Object {
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.571rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1rem",
            "lineHeight": "1.571rem",
          },
        },
        "h1": Object {
          "desktop": Object {
            "fontSize": "2.286rem",
            "lineHeight": "2.571rem",
          },
          "phone": Object {
            "fontSize": "1.714rem",
            "fontWeight": 600,
            "lineHeight": "2rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "2rem",
            "lineHeight": "2.429rem",
          },
        },
        "h2": Object {
          "desktop": Object {
            "fontSize": "1.571rem",
            "lineHeight": "1.857rem",
          },
          "phone": Object {
            "fontSize": "1.429rem",
            "fontWeight": 600,
            "lineHeight": "1.714rem",
            "margin": "0 0 18px 0",
          },
          "tablet": Object {
            "fontSize": "1.429rem",
            "lineHeight": "1.714rem",
          },
        },
        "h3": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 700,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
        "intro": Object {
          "desktop": Object {
            "fontSize": "1.357rem",
            "lineHeight": "1.929rem",
          },
          "phone": Object {
            "fontSize": "1.214rem",
            "fontWeight": 300,
            "lineHeight": "1.714rem",
            "margin": "0 0 36px 0",
          },
          "tablet": Object {
            "fontSize": "1.214rem",
            "lineHeight": "1.714rem",
          },
        },
        "p": Object {
          "desktop": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
          "phone": Object {
            "fontSize": "1.143rem",
            "fontWeight": 400,
            "lineHeight": "1.714rem",
            "margin": "0 0 10px 0",
          },
          "tablet": Object {
            "fontSize": "1.143rem",
            "lineHeight": "1.714rem",
          },
        },
      },
    }
  }
/>
`;

exports[`Test <FilePreview /> should render FilePreview with theme props 1`] = `
<img
  className="sc-bZQynM knJMDX"
/>
`;

exports[`Test <Preview /> should render Preview with theme props 1`] = `
<div
  className="sc-bxivhb cyZIyW"
/>
`;

exports[`Test <PreviewContainer /> should render PreviewContainer with theme props 1`] = `
<div
  className="sc-htpNat hPPCeB"
/>
`;
